src/k8s/pkg/k8sd/app/hooks_bootstrap.go

```diff
@@ -49,15 +48,23 @@ func (a *App) onBootstrapWorkerNode(s *state.State, encodedToken string) error {
 	if nodeIP == nil {
 		return fmt.Errorf("failed to parse node IP address %s", s.Address().Hostname())
 	}
-
 	// TODO(neoaggelos): figure out how to use the microcluster client instead
-	// create an HTTP client that ignores https
-	httpClient := &http.Client{
-		Transport: &http.Transport{
-			TLSClientConfig: &tls.Config{
-				InsecureSkipVerify: true,
-			},
-		},
+
+	// Get remote certificate from the cluster member
+	cert, err := utils.GetRemoteCertificate(nodeIP.String())
```

Comment:
`nodeIP.String()` is not right. First of all, we need a `$host:$port` to connect to, not just an address. Secondly, this must be the address of the remote microcluster node, from `token.JoinAddresses[0]`